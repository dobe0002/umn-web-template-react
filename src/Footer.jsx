import React from 'react';
import MainNavLinks from './HeaderFooter/MainNavLinks.jsx';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const FooterWrapper = styled.footer`
  font-family: Helvetica, 'Helvetica Neue', Futura, Verdana, Geneva, sans-serif;
  padding: 20px 54px;
  text-align: center;
  font-size: 0.8rem;
  color: #000;
  border-top: 1px solid #f0f0f0;
`;

const Link = styled.a`
  color: #404d5b;
  text-decoration: underline;
`;
const Footer = props => {
  return (
    <FooterWrapper role='contentinfo'>
      <MainNavLinks mode='footer' />
      <p>
        &copy; <span className='year'>{props.year}</span> Regents of the
        University of Minnesota. All rights reserved. The University of
        Minnesota is an equal opportunity educator and employer.{' '}
        <Link href='http://privacy.umn.edu'>Privacy Statement</Link>
      </p>
    </FooterWrapper>
  );
};
Footer.propTypes = {
  year: PropTypes.string
};
Footer.defaultProps = {
  year: '2019'
};

export default Footer;
