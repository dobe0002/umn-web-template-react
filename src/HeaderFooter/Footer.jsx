/* eslint-disable import/extensions */
/* eslint-disable react/prop-types */
import React from 'react';
// import styled from 'styled-components'; /* Not using styled-components so  styled-components isn't needed as a dependency */
// import PropTypes from 'prop-types';  /* Disabling this so prop-types isn't needed as a dependency */
import MainNavLinks from './MainNavLinks';

const defaultYear = '2020';
const footerCSS = {
  fontFamily:
    'Helvetica, "Helvetica Neue", Futura, Verdana, Geneva, sans-serif',
  padding: '20px 54px',
  textAlign: 'center',
  fontSize: '0.8rem',
  color: '#000',
  borderTop: '1px solid #f0f0f0',
};
const linkCSS = {
  color: '#404d5b',
  textDecoration: 'underline',
};

const Footer = (props) => {
  const { year, label } = props;
  const displayYear = year === '' || year === undefined ? defaultYear : year;
  const displayLabel =
    label === '' || label === undefined
      ? 'University copyright information'
      : label;
  return (
    <footer role='contentinfo' style={footerCSS} aria-label={displayLabel}>
      <MainNavLinks mode='footer' />
      <p>
        &copy;
        <span className='year'>{displayYear}</span>
        <span> </span>
        Regents of the University of Minnesota. All rights reserved. The
        University of Minnesota is an equal opportunity educator and employer.
        <span> </span>
        <a href='http://privacy.umn.edu' style={linkCSS}>
          Privacy Statement
        </a>
      </p>
    </footer>
  );
};
// Footer.propTypes = {
//   year: PropTypes.string,
//   label: PropTypes.string
// };
// Footer.defaultProps = {
//   year: '2020',
//   label: 'University copyright information'
// };

export default Footer;
