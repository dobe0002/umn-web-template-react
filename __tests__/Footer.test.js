import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Footer from '../src/Footer';
import Adapter from 'enzyme-adapter-react-16';
const { axe, toHaveNoViolations } = require('jest-axe');
expect.extend(toHaveNoViolations);
configure({ adapter: new Adapter() });

describe('UMN Footer', () => {
  it('Accessibility check', async () => {
    const footer = mount(<Footer />);
    const results = await axe(footer.getDOMNode());

    console.log('Axe violations', results.violations);

    expect(results).toHaveNoViolations();
  });
  it('Footer renders correctly', () => {
    const footer = shallow(<Footer year='1234' />);
    expect(footer).toMatchSnapshot();
  });
  it('A default year displayed when year attribute is not passed', () => {
    const footer = shallow(<Footer />);
    const year = footer
      .find('.year')
      .text()
      .trim();
    expect(year).not.toEqual('');
  });
});
