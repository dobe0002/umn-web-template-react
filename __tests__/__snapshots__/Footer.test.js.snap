// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UMN Footer Footer renders correctly 1`] = `
<styled.footer
  role="contentinfo"
>
  <MainNavLinks
    mode="footer"
  />
  <p>
    Â© 
    <span
      className="year"
    >
      1234
    </span>
     Regents of the University of Minnesota. All rights reserved. The University of Minnesota is an equal opportunity educator and employer.
     
    <styled.a
      href="http://privacy.umn.edu"
    >
      Privacy Statement
    </styled.a>
  </p>
</styled.footer>
`;
